yarn-project/acir-simulator/src/avm/avm_context.ts

```diff
@@ -0,0 +1,79 @@
+import { Fr } from '@aztec/foundation/fields';
+
+/**
+ * Store's data for an Avm execution frame
+ */
+export class AvmContext {
+  /** - */
+  public readonly calldata: Fr[];
+  private returnData: Fr[];
```

Comment:
Thoughts on mimicking the structure outlined in [the yellowpaper](https://yp-aztec.netlify.app/docs/public-vm/avm#execution-context) where Context has an ExecutionEnvironment, MachineState, and MessageCallResults?

I'm fine with renaming/restructuring things in the yellowpaper too, but we might want to align the two.

Reply:
I will give this a review and restructure; I think structurally close it just needs a bit of renaming. 
Lets have a walk through / pairing chat about this